doc/development/rfc/version_numbering.md

```diff
@@ -0,0 +1,184 @@
+(a release contains one or more of these changes).
+A point release (or a _dot_ release) can generally apply to anything after
+the first dot, i.e., minor or micro release, while
+using minor and micro does not have that ambiguity.
+Although minor and micro have potential for confusion due to the
+similarity of their names, minor is an established term in this context
+and micro is sometimes used in this context and in other contexts,
+it has the right connotations.
+
+### Development Versions
+
+Development versions have the version number of the next release which will be released
+from a given code base which is defined by a branch.
+
+Development versions of source code on each branch have a dev suffix,
+e.g., 3.5.1-dev.
+Transition to the Semantic Versioning style is strongly recommend
+which means including dash (hyphen) before dev, e.g., 3.5.1-dev.
+Notably, these dev-suffixed version numbers are not unique, i.e.,
+multiple source code versions are marked the same.
+
+Release candidates (RCs) are pre-releases marked by appending RC and
+a release candidate number to the version, e.g., 3.5.1RC2.
+
+Transition to the Semantic Versioning style for pre-releases is strongly recommend.
+Version number is followed by a dash (hyphen) followed by a dot-separated identifier
+which consists of identification of pre-release type, i.e., rc or RC, optional dot,
+and a release candidate number, e.g., 3.5.1-rc.2, 3.5.1-RC.2, 3.5.1-rc2, or 3.5.1-RC2.
+
+Daily builds and other builds of development versions other than release candidates
+may use additional dashes to specify the actual source version, e.g.,
+3.5.1-dev-05e5df2e7.
+
+Any version parsing or build system, must support both systems, the one without
+a dash and the one with dash.
+
+### Build Information
+
+When build information is captured in the version number, Semantic Versioning
+prescribed plus sign as a separator, e.g., 3.5.1+1 or 3.5.1-RC1+1.
+However, filenames with a plus sign may not work well, so a dash as a separator
+is allowed too while keeping in mind that automated semantic version tools
+won't parse the version correctly. A suggested workaround is to use the dash
+only in a file name and elsewhere use plus.
+
+Any version parsing or build system, must support the form with dash
+and the form with plus.
+
+While the assignment of a version is typically done by tagging in a version
+control system, but builds are managed separately.
+
+## Relation to Other Documents
+
+* [RFC 4: Release Procedure](https://trac.osgeo.org/grass/wiki/RFC/4_ReleaseProcedure): No changes to release procedure, only the numbering is changed.
+* [Release Schedule](https://trac.osgeo.org/grass/wiki/Release/Schedule#Externalreleaseschedules) (at Trac wiki under Release): The Release Schedule document describes schedule, branching, release maintenance, and numbering. The numbering is changed, specifically the use of odd version numbers for development.
```

Comment:
Where shall we maintain this schedule in future?

Reply:
Python has release schedules for each minor version as a PEP. Our equivalent would be then a living RFC updated as needed. (The RFC updates are not resolved yet besides the fact that RFC is a misnomer.)

Note: Given the description, the link is meant for the whole schedule, not just the external schedules.

Reply:
Link fixed. Leaving the schedule question for later.